solidity/contracts/deposit/Deposit.sol

```diff
@@ -36,7 +36,6 @@ contract Deposit is DepositFactoryAuthority {
     }
 
     function () external payable {
-        require(msg.sender == self.keepAddress, "Deposit contract can only receive ETH from underlying keep");
```

Comment:
I suggest we check for the `msg.data` length, to avoid allowing any unknown function calls to pass through to the fallback function.

`require(msg.data.length == 0, "Deposit contract was called with unknown function selector.");`

Reply:
> require(msg.data.length == 0, "Deposit contract was called with unknown function selector.");

Oops, this is only the message data. Looks like you should test if `msg.sig == 0x0` or something of that sort. ([ref](https://ethereum.stackexchange.com/questions/61826/how-to-extract-function-signature-from-msg-data)).

Reply:
what do you mean by unknown function calls?

Reply:
See [this error](https://discordapp.com/channels/590951101600235531/707228822445097051/707249615199535185) from Discord and [my explanation](https://discordapp.com/channels/590951101600235531/707228822445097051/707309632929923175). The fallback runs on any unknown method call, not just on a value send. If we `require` that the signature be `0x0`, we can provide a proper revert in cases where the fallback is called because the called method didn't exist, vs simply a value send.

Reply:
> Oops, this is only the message data. Looks like you should test if msg.sig == 0x0 or something of that sort. (ref).

msg.data.length is good :)

Reply:
@NicholasDotSol, tested it, it's not. Try calling an unknown method with no args and value attached. It will not revert.

Reply:
Oh, you're right. Must've stumbled on my end.